lib/import/brca/providers/sheffield/sheffield_handler.rb

```diff
@@ -7,387 +7,533 @@ module Providers
-            #   | unless record.raw_fields['karyotypingmethod'].nil?
-            # unless record.raw_fields['karyotypingmethod'].nil?
-            if BRCA_REGEX.match(record.raw_fields['genotype'])
-              genotype.add_gene($LAST_MATCH_INFO[:brca])
-              @successful_gene_counter += 1
-              @gene_counter += 1
-              @logger.debug "SUCCESSFUL gene parse for: #{$LAST_MATCH_INFO[:brca]}"
-              [genotype]
-            # elsif BRCA_REGEX.match(record.raw_fields['genotype']) and rca_input.strip.scan(BRCA_REGEX).size == 2
-            elsif brcas = extract_brca(record.raw_fields['karyotypingmethod'].to_s)
-              if brcas.size == 1 && karyo != 'BRCA1 and 2 gene sequencing'
-                @logger.debug "JUST ADDED THIS GENE #{brcas}"
-                genotype.add_gene(brcas[0])
-                [genotype]
-              elsif brcas.size > 1
-                @logger.debug "EXTRACTED THESE GENES 2 #{brcas}"
-                genotype2 = genotype.dup
-                genotype.add_gene(7)
-                genotype2.add_gene(8)
-                [genotype, genotype2]
-              elsif (geno == 'BRCA1 and 2 gene analysis' && karyo == 'BRCA1 and 2 gene sequencing') || (geno == 'BRCA1 and 2 gene analysis' && karyo == 'Full Screen')
-                @logger.debug "Found geno + karyo - #{geno} and #{karyo}"
-                genotype2 = genotype.dup
-                genotype.add_gene(7)
-                genotype2.add_gene(8)
-                [genotype, genotype2]
+          def process_single_gene(genotype, record)
+            positive_gene = get_gene(record)
+            genotype.add_gene(positive_gene[0]&.upcase) if positive_gene.present?
+            return if positive_gene.blank?
+
+            @logger.debug "SUCCESSFUL gene parse for: #{positive_gene[0]&.upcase}"
+          end
+
+          def add_other_genes_with_status(other_genes, genotype, genotypes, status)
+            other_genes.each do |gene|
+              genotype_othr = genotype.dup
+              @logger.debug "SUCCESSFUL gene parse for #{status} status for: #{gene}"
+              genotype_othr.add_status(status)
+              genotype_othr.add_gene(gene)
+              genotype_othr.add_protein_impact(nil)
+              genotype_othr.add_gene_location(nil)
+              genotypes.append(genotype_othr)
+            end
+            genotypes
+          end
+
+          def process_multi_genes(genotype, record, genotypes)
+            positive_gene = record.raw_fields['genotype'].scan(BRCA_REGEX).flatten.uniq
+            raw_genotypes = record.raw_fields['genotype'].split(positive_gene[-1])
+            raw_genotypes[1].prepend(positive_gene[-1])
+            process_raw_genotypes(raw_genotypes, genotype, genotypes)
+          end
+
+          def process_raw_genotypes(raw_genotypes, genotype, genotypes)
+            raw_genotypes.each do |raw_genotype|
+              raw_genotype.scan(BRCA_REGEX)
```

Comment:
@shilpigoeldev  the return value of `scan` isn't being assigned to anything, is it being used?

Reply:
@ollietulloch , it is being used at line 307 using  $LAST_MATCH_INFO[:brca] that got populated here to assign the gene.